### DISCUSSION TITLE
Matching multiple paths and identifying sub domains

### DISCUSSION THREAD

**Discussion by bmatatuk** [dotnet/yarp#673] [General] [answered]
We are planning a Angular web application with a partial SaaS model. The static site (SPA) request will need to proxied to a common URL for all the customers. Whereas, the api requests will need to be proxied to their respective API URL for the data (customer will be identified by the sub domain).
When new customers join, they will also be forwarded to their respective API locations. 

I was thinking of having multiple "Paths" within the Match object, but Path is just a single string element. Do I have to create multple route objects for each unique path? 

//---------------------------------------
http://sub1.domain.com/swagger/index.html
http://sub2.domain.com/swagger/index.html
http://sub1.domain.com/
http://sub2.domain.com/
http://www.domain.com/swagger/index.html

Intended Destination: http://www.somedomain.com
//----------------------------------------

//----------------------------------------
http://sub1.domain.com/api/product/list
http://sub1.domain.com/authenticate/token

Intended Destination: http://api.abc.com:8080/
//----------------------------------------

//----------------------------------------
http://sub2.domain.com/api/product/list
http://sub2.domain.com/authenticate/token

Intended Destination: http://www.pqr.com:4334/
//----------------------------------------

Can YARP in its current stage cater for the above requirement? If so, I would appreciate if you could point me to the right direction possibly with an example.

YARP is exactly what I was looking for and can wait to see v1. Thank you

**Comment by Tratcher** [comment, marked as answer]
Worst case you can define multiple routes pointing to the same destination. However, you should be able to reduce the redundancy using wildcards in both the host and path. E.g.:

http://*.domain.com/ -> http://www.somedomain.com
http://*.domain.com/swagger/index.html -> http://www.somedomain.com

In this setup it looks like you'd be programmatically generating the routes? In that case a little redundancy doesn't seem like a big issue, and the route table should scale pretty well.

**Reply by thangchung** [reply]
@Tratcher Do we have a plan to do the merging swagger files on the downstream services into one swagger file on YARP?

**Reply by luizfbicalho** [reply]
This would be a nice feature

**Reply by Tratcher** [reply]
> @Tratcher Do we have a plan to do the merging swagger files on the downstream services into one swagger file on YARP?

That's not something that's come up, but I can see the appeal. YARP is being designed as a pass-through proxy, we don't plan to develop API gateway features like swagger aggregation. However, it should be possible to layer such features on top of the base YARP model.
